Packaging fails when compiled in clean chroot

After disabling the bpftool integration the packaging fails.
This issue was not present in the 1.0.6 release and should be therefore seen as a regression.

Looking on here:
>   Failed to get metadata for config file "/root/.rustfmt.toml": Permission denied (os error 13)

It seems not the respect the cargo_home flags passed to meson:
https://gitlab.archlinux.org/archlinux/packaging/packages/scx-scheds/-/blob/main/PKGBUILD?ref_type=heads#L39


``` 
==> Starting package()...
ninja: Entering directory `/build/scx-scheds/src/scx/build'
[1/45] Generating libbpf with a custom command

[2/35] Generating bpftool_target with a custom command
[3/11] Generating from 'lib.bpf.o'
[4/11] Generating from 'lib.bpf.o'
[5/11] Generating from 'lib.bpf.o'
[6/11] Generating from 'lib.bpf.o'
[7/11] Generating from 'lib.bpf.o'
[8/11] Generating from 'lib.bpf.o'
[9/11] Generating from 'lib.bpf.o'
[10/11] Generating from 'lib.bpf.o'
[11/11] Generating rust_all with a custom command (wrapped by meson to set env)
FAILED: Cargo.toml.__PHONY__ 
/usr/bin/meson --internal exe --unpickle /build/scx-scheds/src/scx/build/meson-private/meson_exe_cargo_67f70dec912b7fc7bd1c3826682f5a35edafb437.dat
while executing ['/usr/bin/cargo', 'build', '--manifest-path=../Cargo.toml', '--target-dir=.', '--quiet', '--release']
--- stdout ---

--- stderr ---
warning: creating a shared reference to mutable static is discouraged
   --> rust/scx_rustland_core/src/alloc.rs:457:19
    |
457 |             match VM.save() {
    |                   ^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives
    = note: `#[warn(static_mut_refs)]` on by default

warning: creating a shared reference to mutable static is discouraged
   --> rust/scx_rustland_core/src/alloc.rs:482:19
    |
482 |             match VM.restore() {
    |                   ^^^^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives

warning: creating a shared reference to mutable static is discouraged
   --> rust/scx_rustland_core/src/alloc.rs:525:54
    |
525 |     unsafe { UserAllocator::new(BuddyAllocParam::new(HEAP.0.as_ptr(), HEAP_SIZE, LEAF_SIZE)) };
    |                                                      ^^^^^^^^^^^^^^^ shared reference to mutable static
    |
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html>
    = note: shared references to mutable statics are dangerous; it's undefined behavior if the static is mutated or if a mutable reference is created for it while the shared reference lives

error: failed to run custom build command for `scx_rustland v1.0.7 (/build/scx-scheds/src/scx/scheds/rust/scx_rustland)`

Caused by:
  process didn't exit successfully: `/build/scx-scheds/src/scx/build/./release/build/scx_rustland-c8fe1ae2d6f1e349/build-script-build` (exit status: 101)
  --- stdout
  scx_utils:clang=ClangInfo { clang: "/usr/bin/clang", ver: "18.1.8", arch: "x86_64" } ["-g", "-O2", "-Wall", "-Wno-compare-distinct-pointer-types", "-D__TARGET_ARCH_x86", "-mcpu=v3", "-mlittle-endian", "-idirafter", "/usr/lib/clang/18/include", "-idirafter", "/usr/local/include", "-idirafter", "/usr/include", "-I/build/scx-scheds/src/scx/build/./release/build/scx_rustland-c36a4de510598e93/out/scx_utils-bpf_h/arch/x86", "-I/build/scx-scheds/src/scx/build/./release/build/scx_rustland-c36a4de510598e93/out/scx_utils-bpf_h", "-I/build/scx-scheds/src/scx/build/./release/build/scx_rustland-c36a4de510598e93/out/scx_utils-bpf_h/bpf-compat"]
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=intf.h
  cargo:rerun-if-changed=/usr/lib/clang/18/include/stdbool.h

  --- stderr
  Failed to get metadata for config file "/root/.rustfmt.toml": Permission denied (os error 13)
  Failed to run rustfmt: Internal rustfmt error (non-fatal, continuing)
  Warning: unrecognized map: license
  Warning: unrecognized map: .maps
  Warning: unrecognized map: .ksyms
  Failed to get metadata for config file "/root/.rustfmt.toml": Permission denied (os error 13)
  thread 'main' panicked at scheds/rust/scx_rustland/build.rs:8:10:
  called `Result::unwrap()` on an `Err` value: failed to generate `/build/scx-scheds/src/scx/build/./release/build/scx_rustland-c36a4de510598e93/out/bpf.bpf.o`

  Caused by:
      0: Failed to generate skeleton for /build/scx-scheds/src/scx/build/./release/build/scx_rustland-c36a4de510598e93/out/bpf.bpf.o
      1: Failed to rustfmt: exit status: 1
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: failed to run custom build command for `scx_rlfifo v1.0.7 (/build/scx-scheds/src/scx/scheds/rust/scx_rlfifo)`

Caused by:
  process didn't exit successfully: `/build/scx-scheds/src/scx/build/./release/build/scx_rlfifo-d7d055d470c98909/build-script-build` (exit status: 101)
  --- stdout
  scx_utils:clang=ClangInfo { clang: "/usr/bin/clang", ver: "18.1.8", arch: "x86_64" } ["-g", "-O2", "-Wall", "-Wno-compare-distinct-pointer-types", "-D__TARGET_ARCH_x86", "-mcpu=v3", "-mlittle-endian", "-idirafter", "/usr/lib/clang/18/include", "-idirafter", "/usr/local/include", "-idirafter", "/usr/include", "-I/build/scx-scheds/src/scx/build/./release/build/scx_rlfifo-308282fa5eaf7ee0/out/scx_utils-bpf_h/arch/x86", "-I/build/scx-scheds/src/scx/build/./release/build/scx_rlfifo-308282fa5eaf7ee0/out/scx_utils-bpf_h", "-I/build/scx-scheds/src/scx/build/./release/build/scx_rlfifo-308282fa5eaf7ee0/out/scx_utils-bpf_h/bpf-compat"]
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=intf.h
  cargo:rerun-if-changed=/usr/lib/clang/18/include/stdbool.h

  --- stderr
  Failed to get metadata for config file "/root/.rustfmt.toml": Permission denied (os error 13)
  Failed to run rustfmt: Internal rustfmt error (non-fatal, continuing)
  Warning: unrecognized map: license
  Warning: unrecognized map: .maps
  Warning: unrecognized map: .ksyms
  Failed to get metadata for config file "/root/.rustfmt.toml": Permission denied (os error 13)
  thread 'main' panicked at scheds/rust/scx_rlfifo/build.rs:8:10:
  called `Result::unwrap()` on an `Err` value: failed to generate `/build/scx-scheds/src/scx/build/./release/build/scx_rlfifo-308282fa5eaf7ee0/out/bpf.bpf.o`

  Caused by:
      0: Failed to generate skeleton for /build/scx-scheds/src/scx/build/./release/build/scx_rlfifo-308282fa5eaf7ee0/out/bpf.bpf.o
      1: Failed to rustfmt: exit status: 1
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

ninja: build stopped: subcommand failed.
Could not rebuild /build/scx-scheds/src/scx/build

```

--- a/rustfmt.toml
+++ b/rustfmt.toml
@@ -1,0 +1,3 @@
+# this file is here to prevent build breakages on arch
+edition = "2021"
+

